### DISCUSSION TITLE
Gagal Run App Shinny dashboard

### DISCUSSION THREAD

**Discussion by DeanAlexander27** [teamalgoritma/community#563] [Q&A] [answered]
Dear Ka Instruktur 
mohon bantuannya terdapat error sebagai berikut. Apakah ada kesalahan pengetikan ?

![image](https://user-images.githubusercontent.com/123348110/220559955-d989492d-907f-4d22-ab17-6122a2b456e6.png)

berikut kodingan pada sheet server.R :

function(input, output) { 
  
  output$progress2022 <- renderPlotly({
    
    datasetsales <- dataset_long %>% 
      filter(year == input$input1) %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="Sales")
    
    datasetsales1 <- datasetsales %>% 
      select(key_figures,total,year,month) %>% 
      arrange(year, month) 
    
    datasettotalstock <- dataset_long %>%
      filter(year == input$input1) %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="Total Stock")
    
    datasettotalstock1 <- datasettotalstock %>% 
      select(key_figures,total,year,month) %>% 
      arrange(year, month) 
    
    datasetmandist <- dataset_long %>% 
      filter(year == input$input1) %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="MANUF DISTRIBUTION")
    
    datasettotalstock1 <- datasettotalstock1 %>% 
      mutate(label=glue("Total Stock {month} {year} :
                    {total} unit"))
    
    datasetsales1 <- datasetsales1 %>% 
      mutate(label = glue ("Total Sales {month} {year} :
                       {total} unit "))
    
    datasetmandist1 <- datasetmandist1 %>% 
      mutate(label = glue ("Total Manufacturing Distribution
                       {month} {year} : {total} unit"))
    
    
    plot1 <- ggplot(data = datasettotalstock1, aes(x = month, y = total, fill = as.factor(year), text = label)) +
      geom_col(position = 'dodge') +
      labs(title = "Progress Total Stock, Manufacturing Distribution, Sales 2019 - 2022")+
      scale_fill_manual(name = NULL, values = c('#d0cdd7', '#acb0bd', '#416165', '#0b3948')) +
      theme_algoritma
    
    plot2 <- ggplot(data = datasetmandist1, aes(x = month, y = total, color = factor(year), group = year, text = label)) +
      geom_line(size = 1) +
      geom_point() +
      scale_color_brewer(palette = "Set3") +
      theme_algoritma
    
    plot3 <- ggplot(data = datasetsales1, aes(x = month, y = total, color = factor(year), group = year, text = label)) +
      geom_line(size = 1) +
      geom_point() +
      scale_colour_manual(values = c("#fe8181", "#fe5757", "#fe2e2e", "#b62020")) +
      theme_algoritma
    
    # combine plots using subplot function
    progress2022<- subplot(
      ggplotly(plot1, tooltip = "text"),
      ggplotly(plot2, tooltip = "text"),
      ggplotly(plot3, tooltip = 'text'),
      nrows = 3,
      margin = 0.05
    )
    
    
    
    
  })
  
  }

**Comment by ranydwi** [comment, marked as answer]
Dear @DeanAlexander27 
Terima kasih untuk pertanyaan yang masuk. 

Untuk error yang bapak hadapi, apabila saya lihat dari code server Bapak, sepertinya dikarenakan Bapak belum memanggil object `progress2022` berupa subplot yang berisikan visualisasi interaktif. Silahkan panggil object tersebut setelah kurung penutup `margin = 0.05`. 
```
# combine plots using subplot function
progress2022<- subplot(
  ggplotly(plot1, tooltip = "text"),
  ggplotly(plot2, tooltip = "text"),
  ggplotly(plot3, tooltip = 'text'),
  nrows = 3,
  margin = 0.05
)
progress2022
```

Kemudian coba untuk running kembali. Apakah masih terdapat error yang sama/tidak.

Semoga membantu :)

**Reply by DeanAlexander27** [reply]
halo.. terima kasih ka.. tetapi masih error ka dengan argumen 1 not a vector..
apa ada solusi lain ?

berikut koding nya 

library(shiny)
library(shinydashboard)

ui <- dashboardPage(
  
  dashboardHeader(title = "PT.A Supply Chain Division"),
  
  dashboardSidebar(
    sidebarMenu(
      menuItem("Dashboard", tabName = "dashboard", icon = icon("dashboard")),
      menuItem("Distribution Progress", icon = icon("th"), tabName = "widgets",
               badgeLabel = "new", badgeColor = "green"),
      menuItem("Data",icon = icon("users"),tabName = "widgets",
               badgeLabel = "new", badgeColor = "green"),
      selectInput(inputId = "input1", 
                  label = "Choose Year",
                  choices = levels(datasettotalstock$year))
    )
  ),
  
  dashboardBody(
    tabItems(
      tabItem(tabName = "dashboard",
              h2("Supply Chain Dashboard", align ="center"),
              br(),
              fluidRow(
                infoBox(title = "Total Sales July 2022", value = tail(datasetsales1$total_sales,1), icon = icon("credit-card"), fill = TRUE,color='red'),
                infoBox(title = "Distribution Stock 2022", value = tail(datasetmandistnew1$total_dist,1), icon = icon("truck"),fill = TRUE,color='orange'),
                infoBox(title = "Total Stock 2022", 2685, icon = icon("boxes-stacked"),fill = TRUE)
              ),
              
                plotlyOutput(outputId = "progress2022")
                    
              ),
      
      tabItem(tabName = "widgets",
              h2("Widgets tab content")),
      
      tabItem(tabName = "widgets",
              h2("Widgets tab content"))
   )
)
)

**Reply by DeanAlexander27** [reply]
untuk server.R ka @ranydwi 

function(input, output) { 
  
  output$progress2022 <- renderPlotly({
    
    datasetsales <- dataset1 %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="Sales") %>% 
      filter(year == input$input1)
    
    datasettotalstock <- dataset1 %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="Total Stock") %>% 
      filter(year == input$input1)
    
    datasetmandist <- dataset1 %>% 
      group_by(key_figures,year,month) %>% 
      summarise(total = sum(value)) %>% 
      filter(key_figures=="MANUF DISTRIBUTION") %>% 
      filter(year == input$input1)
    
    datasettotalstock <- datasettotalstock %>% 
      mutate(label=glue("Total Stock {month} {year} :
                    {total} unit"))
    
    datasetsales <- datasetsales %>% 
      mutate(label = glue ("Total Sales {month} {year} :
                       {total} unit "))
    
    datasetmandist <- datasetmandist %>% 
      mutate(label = glue ("Total Manufacturing Distribution
                       {month} {year} : {total} unit"))
    
    
    plot1 <- ggplot(data = datasettotalstock, aes(x = month, y = total, fill = as.factor(year), text = label)) +
      geom_col(position = 'dodge') +
      labs(title = "Progress Total Stock, Manufacturing Distribution, Sales 2019 - 2022")+
      scale_fill_manual(name = NULL, values = c('#d0cdd7', '#acb0bd', '#416165', '#0b3948')) +
      theme_algoritma
    
    plot2 <- ggplot(data = datasetmandist, aes(x = month, y = total, color = factor(year), group = year, text = label)) +
      geom_line(size = 1) +
      geom_point() +
      scale_color_brewer(palette = "Set3") +
      theme_algoritma
    
    plot3 <- ggplot(data = datasetsales, aes(x = month, y = total, color = factor(year), group = year, text = label)) +
      geom_line(size = 1) +
      geom_point() +
      scale_colour_manual(values = c("#fe8181", "#fe5757", "#fe2e2e", "#b62020")) +
      theme_algoritma
    
    # combine plots using subplot function
    progress2022<- subplot(
      ggplotly(plot1, tooltip = "text"),
      ggplotly(plot2, tooltip = "text"),
      ggplotly(plot3, tooltip = 'text'),
      nrows = 3,
      margin = 0.05
    )
    progress2022
    
    
    
  })
  
  }
